ExtraDry/ExtraDry.Server/RuleEngine.cs

```diff
@@ -147,17 +147,28 @@ private async Task ProcessIndividualUpdate<T>(RuleAction action, PropertyInfo pr
             }
         }
         else {
-            var same = (result == null && destinationValue == null) || (result?.Equals(destinationValue) ?? false);
+            bool same = AreEqual(result, destinationValue);
             if(action == RuleAction.Block && !same) {
-                if(ignore?.Condition == JsonIgnoreCondition.Always) {
+                if(ignoreAttribute?.Condition == JsonIgnoreCondition.Always) {
                     return;
                 }
                 throw new DryException($"Invalid attempt to change property '{property.Name}'", $"Attempt to change read-only property '{property.Name}'");
             }
+            // Do not allow property to be set to the value configured in DeleteValue.
```

Comment:
Couple of suggestions
1. If we check for the property type = Enum and throw an exception - this is limited in its usefulness since it does not look at the other properties that have a DeleteValue set.
2. We consider all properties that have a DeleteValue set and throw an exception if all of these match. (This means the object is not soft deleted unless all the property values match the DeleteValues set)